java/src/jmri/swing/NamedBeanComboBox.java

```diff
@@ -454,7 +454,7 @@ public NamedBeanEditor(ComboBoxEditor editor) {
                 jc.setInputVerifier(new JInputValidator(jc, true, false) {
                     @Override
                     protected Validation getValidation(JComponent component, JInputValidatorPreferences preferences) {
-                        if (component instanceof JTextComponent) {
+                        if (component instanceof JTextComponent && getSelectedIndex() != -1) {
```

Comment:
This is not the correct fix because on line 464 in this method, we set the selected item and therefor selected index. If we stipulate that the selected index can only change if a prior selection has been made, keyboard input will not be able to move the selection in a non-editable combobox, which breaks a design intention (i.e. the intention is that, even in a non-editable combobox, a user should be able to begin typing a name and have the combobox scroll to the closest match; breaking that will be considered a regression).